frontend/public/components/overview/_overview.scss

```diff
@@ -1,38 +1,84 @@
+.co-m-nav-title--overview {
+  @media (min-width: $screen-md-min) {
+    flex-direction: row;
+  }
+
+  // stack form fields when sidebar is open, except when > 1200
+  .overview--sidebar-shown & {
+    flex-direction: column;
+    @media (min-width: $screen-lg-min) {
+      flex-direction: row;
+    }
+  }
+}
+
 .overview-toolbar {
   margin: 0 0 ($grid-gutter-width / 2) 0;
   min-width: 0;
   padding: 0;
-  @media (max-width: $screen-sm-max) {
-    // maintain single row alignment
-    .toolbar-pf-action-right {
-      align-items: flex-end;
-      display: flex;
-      float: none;
-      justify-content: space-between;
+
+  .toolbar-pf-actions {
+    margin: 0;
+  }
+
+  // Override pf and make toolbar a flex object to control positioning
+  .toolbar-pf-action-right {
+    align-items: flex-end;
+    display: flex;
+    float: none;
+
+    @media (min-width: $screen-sm-min) and (max-width: $screen-md-max) {
+      // switch to stacked fields when side open but still at desktop widths
+      .overview--sidebar-shown & {
+        align-items: initial;
+        display: flex;
+        flex-direction: column;
+        float: none;
+        width: calc(100% - 535px);
+        min-width: 160px;
+      }
     }
   }
 }
 
 .overview-toolbar__dropdown {
   display: inline-block;
   max-width: 100%; // enable text-overflow: ellipsis
+
   @media (max-width: $screen-sm-max) {
     min-width: 100%;
   }
+
+  .overview--sidebar-shown & {
+    @media (max-width: $screen-md-max) {
+      min-width: 100%; // stack selections
+    }
+  }
+
   .btn-dropdown {
-    max-width: 250px; // same at default .text-filter
     width: 100%;
+  }
 
-    @media (min-width: $screen-xs-min) and (max-width: $screen-sm-max) {
-      max-width: 100%;
-    }
+  .btn-link__titlePrefix {
+    color: lighten($btn-default-color, 15%);
+    font-weight: normal;
+    margin-right: 2px;
+  }
+
+  .dropdown-menu {
+    min-width: 250px;
   }
 }
 
 .overview-toolbar__form-group {
-  flex: 1 1 50%;
+  flex: 1 1 auto;
+
+  @media (max-width: $screen-sm-max) {
```

Comment:
Nit:  I'd not disrupt the declarations with the override (i.e., move this block to line 80).